### PR TITLE
Add early --testnet medalla support

### PR DESCRIPTION
**Pull request by dapplion**
Adds the option `--testnet medalla` to the CLI. It required some changes aside from adding its particular spec params:

- Type testnet config object. I've removed values that are the same as current defaults. Also, `api.rest.enabled` had an incorrect json path.
- Make `genesisStateFile` and `eth1.enabled` optional to the genesis state file URL existing. It currently doesn't for Medalla, so eth1 will be enabled.

Other goodies

- Call to action in docs home

Screenshot from 2020-07-31 18-57-43

### CODE REVIEW

File: packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      enabled: true,
+      bindAddr: "/ip4/0.0.0.0/udp/9000",
+      bootEnrs: [
+        "enr:-LK4QKWk9yZo258PQouLshTOEEGWVHH7GhKwpYmB5tmKE4eHeSfman0PZvM2Rpp54RWgoOagAsOfKoXgZSbiCYzERWABh2F0dG5ldHOIAAAAAAAAAACEZXRoMpAAAAAAAAAAAAAAAAAAAAAAgmlkgnY0gmlwhDQlA5CJc2VjcDI1NmsxoQOYiWqrQtQksTEtS3qY6idxJE5wkm0t9wKqpzv2gCR21oN0Y3CCIyiDdWRwgiMo",
+        "enr:-LK4QEnIS-PIxxLCadJdnp83VXuJqgKvC9ZTIWaJpWqdKlUFCiup2sHxWihF9EYGlMrQLs0mq_2IyarhNq38eoaOHUoBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpAAAAAAAAAAAAAAAAAAAAAAgmlkgnY0gmlwhA37LMaJc2VjcDI1NmsxoQJ7k0mKtTd_kdEq251flOjD1HKpqgMmIETDoD-Msy_O-4N0Y3CCIyiDdWRwgiMo",
+        "enr:-KG4QIOJRu0BBlcXJcn3lI34Ub1aBLYipbnDaxBnr2uf2q6nE1TWnKY5OAajg3eG6mHheQSfRhXLuy-a8V5rqXKSoUEChGV0aDKQGK5MywAAAAH__________4JpZIJ2NIJpcIQKAAFhiXNlY3AyNTZrMaEDESplmV9c2k73v0DjxVXJ6__2bWyP-tK28_80lf7dUhqDdGNwgiMog3VkcIIjKA"
+      ],
+    },
+    maxPeers: 25,
+    bootMultiaddrs: [],
```
Comment (dapplion) [inline packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts:38]: Are there any `bootMultiaddrs` for Medalla or only ENRs?

File: packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      enabled: true,
+      bindAddr: "/ip4/0.0.0.0/udp/9000",
+      bootEnrs: [
+        "enr:-LK4QKWk9yZo258PQouLshTOEEGWVHH7GhKwpYmB5tmKE4eHeSfman0PZvM2Rpp54RWgoOagAsOfKoXgZSbiCYzERWABh2F0dG5ldHOIAAAAAAAAAACEZXRoMpAAAAAAAAAAAAAAAAAAAAAAgmlkgnY0gmlwhDQlA5CJc2VjcDI1NmsxoQOYiWqrQtQksTEtS3qY6idxJE5wkm0t9wKqpzv2gCR21oN0Y3CCIyiDdWRwgiMo",
+        "enr:-LK4QEnIS-PIxxLCadJdnp83VXuJqgKvC9ZTIWaJpWqdKlUFCiup2sHxWihF9EYGlMrQLs0mq_2IyarhNq38eoaOHUoBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpAAAAAAAAAAAAAAAAAAAAAAgmlkgnY0gmlwhA37LMaJc2VjcDI1NmsxoQJ7k0mKtTd_kdEq251flOjD1HKpqgMmIETDoD-Msy_O-4N0Y3CCIyiDdWRwgiMo",
+        "enr:-KG4QIOJRu0BBlcXJcn3lI34Ub1aBLYipbnDaxBnr2uf2q6nE1TWnKY5OAajg3eG6mHheQSfRhXLuy-a8V5rqXKSoUEChGV0aDKQGK5MywAAAAH__________4JpZIJ2NIJpcIQKAAFhiXNlY3AyNTZrMaEDESplmV9c2k73v0DjxVXJ6__2bWyP-tK28_80lf7dUhqDdGNwgiMog3VkcIIjKA"
+      ],
+    },
+    maxPeers: 25,
+    bootnodes: [],
```
Comment (dapplion) [inline packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts:38]: Are there any multiaddress bootnodes for Medalla?

File: docs/usage/medalla.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+2020-06-21 17:35:31  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xcc1eb4c83cb0d9a9f7ed0045d2cd6257aadd226420baabd3d672be35605fe470, slot=112, epoch=3
+2020-06-21 17:35:31  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xb5c7b08a369c0943292a91e733f3feed184bc3c0b49d04878bc86e0705c15fe8, slot=113, epoch=3
+```
+
+<!-- prettier-ignore-start -->
+!!! info
+    If your node is stuck with `Current peerCount=0` review your network configuration to make sure your ports are open.
+<!-- prettier-ignore-end -->
+
+A young testnet (such as Altona) should take a few hours to sync. If you see multiple or consistent errors in the logs, please open a [Github issue](https://github.com/ChainSafe/lodestar/issues/new) or reach out to us in [Discord](https://discord.gg/yjyvFRP). Just by reporting anomalities you are helping accelerate the progress of Eth2.0, thanks for contributing!
```
Comment (mpetrunic) [inline docs/usage/medalla.md]: such as Medalla?

File: packages/lodestar-cli/src/cmds/beacon/testnets/altona.ts [hunk 4]
```diff
@@ -1,50 +1,34 @@
+import {IBeaconNodeOptionsPartial} from "../../../options";
+
 /* eslint-disable max-len */
 
 // Use a Typescript file instead of JSON so it's automatically included in the built files
 
-export const altonaConfig = {
-  "api": {
-    "enabled": true,
-    "namespaces": ["beacon", "validator"],
```
Comment (mpetrunic) [inline packages/lodestar-cli/src/cmds/beacon/testnets/altona.ts:8]: not sure where are default values, but "node" should be included here

File: packages/lodestar-cli/src/cmds/beacon/testnets/altona.ts [hunk 5]
```diff
@@ -1,50 +1,34 @@
...
-      "origin": "*"
-    }
-  },
-  "eth1": {
-    "enabled": false,
-    "provider": {
-      "url": "http://goerli.prylabs.net"
+export const altonaConfig: IBeaconNodeOptionsPartial = {
+  api: {
+    rest: {
```
Comment (3xtr4t3rr3str14l) [inline packages/lodestar-cli/src/cmds/beacon/testnets/altona.ts:9]: I have the same question here as I do in this PR:
https://github.com/ChainSafe/lodestar/pull/1267#discussion_r463761752

File: packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import {IBeaconNodeOptionsPartial} from "../../../options";
+
+/* eslint-disable max-len */
+
+// Use a Typescript file instead of JSON so it's automatically included in the built files
+
+export const medallaConfig: IBeaconNodeOptionsPartial = {
+  api: {
+    rest: {
```
Comment (3xtr4t3rr3str14l) [inline packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts:9]: (same as above comment)

File: packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts (hunk 2 above)
Comment (wemeetagain) [reply packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts:38]: https://github.com/goerli/medalla/blob/master/medalla/bootnodes.txt

Comment (dapplion) [reply packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts:38]: Those are already included as default bootnode ENRs. Eventually, https://github.com/ChainSafe/lodestar/issues/1226 should fix the confusing names

File: packages/lodestar-cli/src/cmds/beacon/testnets/altona.ts (hunk 4 above)
Comment (dapplion) [reply packages/lodestar-cli/src/cmds/beacon/testnets/altona.ts:8]: Defaults can be found here
https://github.com/ChainSafe/lodestar/blob/c382969c82ddc821dfc899368ac1efd5ebb8ac62/packages/lodestar/src/api/rest/options.ts#L13

File: packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts (hunk 2 above)
Comment (wemeetagain) [reply packages/lodestar-cli/src/cmds/beacon/testnets/medalla.ts:38]: oh, right, sorry, misread the comment. I don't know of any multiaddr bootnodes

**Review by wemeetagain — DISMISSED** [review]

**Comment by wemeetagain** [comment]
file conflict

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**